pyomo/solvers/plugins/solvers/cplex_direct.py

```diff
@@ -105,13 +105,16 @@ def _apply_solver(self):
             for block in self._pyomo_model.block_data_objects(descend_into=True, active=True):
                 for var in block.component_data_objects(ctype=pyomo.core.base.var.Var, descend_into=False, active=True, sort=False):
                     var.stale = True
+        _log_file = self._log_file
+        if (self.version()[0] > 12) or (self.version()[0] == 12 and self.version()[1] >= 10):
+            _log_file = open(self._log_file, 'w')
```

Comment:
As a general question, can we do this in a `try ... finally` block so that we can guarantee that the _log_file is closed (even in the case of exception)?

Reply:
@jsiirola Good suggestion